### PR TITLE
Update to be FARGATE compatible

### PR DESCRIPTION
**Pull request by SaiPrasannaGopularam**
# Issue
- Throws error when changing the task launch type from EC2 to FARGATE 

# Changes
- Update to make this module compatible to create FARGATE launch type tasks when required
- Add requires_compatibilities, cpu, memory, execution_role_arn attributes to task_definition block which are required for fargate 
- Terraform fmt

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -90,6 +90,21 @@ variable "network_mode" {
...
 }
 
+variable "req_compatibilities" {
+  description = "Launch type required by the task. Either EC2 or FARGATE"
+  default     = "[EC2]"
+}
+
+variable "cpu" {
+  description = "Number of cpu units used by the task. Required for FARGATE type"
+  default     = ""
```
Comment (automationforthepeople) [inline variables.tf:100]:
```suggestion
  default     = null
```

File: variables.tf [hunk 2]
```diff
@@ -90,6 +90,21 @@ variable "network_mode" {
...
+}
+
+variable "memory" {
+  description = "Amount (in MiB) of memory used by the task. Required for FARGATE type"
+  default     = ""
```
Comment (automationforthepeople) [inline variables.tf:105]:
```suggestion
  default     = null
```

File: ecs.tf [hunk 3]
```diff
@@ -22,10 +22,14 @@ data "template_file" "container_definition" {
...
-  container_definitions = data.template_file.container_definition.rendered
-  network_mode          = var.network_mode
-  task_role_arn         = aws_iam_role.task.arn
+  family                   = "${var.service_identifier}-${var.task_identifier}"
+  container_definitions    = data.template_file.container_definition.rendered
+  network_mode             = var.network_mode
+  requires_compatibilities = var.req_compatibilities
+  cpu                      = var.cpu
+  memory                   = var.memory
+  execution_role_arn       = aws_iam_role.task.arn
```
Comment (automationforthepeople) [inline ecs.tf:31]: `execution_role_arn` and `task_role_arn` are reusing the same role?

Comment (jamiepedwards) [reply ecs.tf:31]: Our standard is to keep these two roles separate with the permissions scoped appropriately.

Comment (automationforthepeople) [reply ecs.tf:31]: @SaiPrasannaGopularam This composable terraform makes me want to keep it simple. You should test this does what you want and expect before merging and making a mess of the commit history

Comment (SaiPrasannaGopularam) [reply ecs.tf]: Created an execution role separately and tested whether it creates a fargate task
https://app.circleci.com/pipelines/github/FitnessKeeper/terraform-payments/50/workflows/a8ec5211-1ecc-4869-90d9-61501045e3fd/jobs/227

**Review by automationforthepeople — APPROVED** [review]

File: variables.tf [hunk 4]
```diff
@@ -153,6 +168,11 @@ variable "app_port" {
   description = "Numeric port on which application listens (unnecessary if neither alb_enable_https or alb_enable_http are true)"
 }
 
+variable "host_port" {
+  description = "Numeric port on which you want to map it to on the host"
+  default = 0
```
Comment (soop3rn0t) [inline variables.tf:173]: is this a reasonable default?

**Pull request merged**